### ISSUE TITLE
Refactor

### ISSUE THREAD

**Issue by matteosuppo** [arduino/arduino-create-agent#3]
This project would probably benefit from a refactor.

We could organize the code in different packages:

**main**: starts things up, manage the daemonization
**server**: manage the http and websocket server
`StartServer`: start a http and https server and serves the requests
**discovery**: searches for serial and network ports
`Loop`: starts a Loop updating the list of ports found
**monitor**: dialogues with serial and network ports
`Open`: open a port to communicate, returns a channel where you can push info
`Close`: close a port to stop communicating
**program**: upload a sketch on a board connected to a port
`Program`: program a port

**Comment by AlbyIanna** [comment]
@matteosuppo @mastrolinux Is there any plan to proceed on this refactor? It would be really useful for the frontend, since the logic to handle this stuff is crazy.

**Comment by mastrolinux** [comment]
Yes, most of these features are now in the arduino-cli project. We should be able to integrate them here within the next 2 months.

**Comment by AlbyIanna** [comment]
Cool. Maybe we should migrate to jira issues then.
